### PR TITLE
Scripts/Spells: Hunter spell Exhilaration

### LINKED ISSUE #19605: Core/Script The spell Exhilaration has an issue.
**Description:**

The spell Exhilaration [Spell ID: 109304] of the Hunter class, has an issue.

**Current behaviour:**

The spell Exhilaration is only healing the Hunter and not the pet aswell .

**Expected behaviour:**

Once you use the spell Exihilaration it should heal the Hunter for 30% health and the Hunter's pet for 100% maximum health.

**Steps to reproduce the problem:**

1. Create a Hunter character.
2. Reach level 48 in order to learn the spell ''Exihilaration'' (rank 2) [Spell ID: 109304].
3. Use ''Exihilaration'' [Spell ID: 109304] while you and your pet are both below 90% health or lower.

**Branch(es):**

Master

**TC rev. hash/commit:** 

7c66e7bbc7a401e7a2d8f2bdcead6531678262ea

**TDB version:**  7.2.0+updates

**Operating system:** Windows 10

Description in wowhead http://www.wowhead.com/spell=109304/exhilaration

### PR DESCRIPTION
**Pull request by MartGon**
**Changes proposed:**

-  Simple fix for Exhilaration spell http://www.wowhead.com/spell=109304/exhilaration
with Exhilaration (Rank 2) http://www.wowhead.com/spell=231546/exhilaration

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:** 
Closes #19605

**Tests performed:** 
Tested in-game on the following scenarios:
- Without Rank 2 the spell only heals the hunter.
- Tested with Rank 2 spell heals both hunter and pet.
- The spell does not glitch if pet is dead or despawned.

**Known issues and TODO list:**
None as far as I know.

### CODE REVIEW

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 1]
```diff
@@ -264,6 +265,43 @@ class spell_hun_disengage : public SpellScriptLoader
...
+                ({
+                    SPELL_HUNTER_EXHILARATION_R2
+                });
+            }
+
+            void HandleOnHit()
+            {
+                if (GetCaster()->HasAura(SPELL_HUNTER_EXHILARATION_R2))
+                    if (Unit* pet = GetCaster()->GetGuardianPet())
+                        GetCaster()->CastSpell(pet, SPELL_HUNTER_EXHILARATION_PET, true);
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_hunter.cpp]: this spell targets the pet implicitly

Comment (MartGon) [inline src/server/scripts/Spells/spell_hunter.cpp]: Mmm I don't get what you mean. Do you mean that it should be casted by the pet?

Comment (Keader) [inline src/server/scripts/Spells/spell_hunter.cpp]: even if casted by player, will hit pet

Comment (MartGon) [inline src/server/scripts/Spells/spell_hunter.cpp]: So there's no need on checking if the pet exists?

**Comment by MartGon** [comment]
I made the checks I did before and it works like the same. So I suppose this solution is better.

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 2]
```diff
@@ -264,6 +265,42 @@ class spell_hun_disengage : public SpellScriptLoader
...
+                return ValidateSpellInfo
+                ({
+                    SPELL_HUNTER_EXHILARATION_R2
+                });
+            }
+
+            void HandleOnHit()
+            {
+                if (GetCaster()->HasAura(SPELL_HUNTER_EXHILARATION_R2))
+                    GetCaster()->CastSpell((Unit*)0, SPELL_HUNTER_EXHILARATION_PET, true);
```
Comment (Keader) [inline src/server/scripts/Spells/spell_hunter.cpp]: GetCaster()->CastSpell((Unit*)nullptr, SPELL_HUNTER_EXHILARATION_PET, true);

**Comment by MartGon** [comment]
I used (Unit*)0 because it was the default value on CastCustomSpell, never used null pointer on target before.

**Comment by MartGon** [comment]
With the last commit, now the code it is as Keader said, in case my last comment was confusing

**Comment by Keader** [comment]
@MartGon has others examples in core: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp#L125 xD

**Comment by MartGon** [comment]
@Keader Thanks for the example! Now I undestand what you said, it is null target because the spell already targets the pet by itself by the target field = TARGET_UNIT_PET. But what I was trying to mean is that I think the PR is ready to merge, if you like of course xD

**Comment by Keader** [comment]
can be merged @joschiwald ?

**Review by Shauren — APPROVED** [review]

**Pull request closed without merging**